components/script/dom/htmlcollection.rs

```diff
@@ -235,7 +235,7 @@ impl HTMLCollection {
     pub fn elements_iter_before<'a>(&'a self, before: &'a Node) -> impl Iterator<Item=Root<Element>> + 'a {
         // Iterate backwards from a node.
         HTMLCollectionElementsIter {
-            node_iter: before.preceding_nodes(&self.root),
+            node_iter: before.preceding_nodes::<Node>(&self.root),
```

Comment:
This type annotation is useless, given `node_iter` must be an iterator that yields `Root<Node>` values.